crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java

```diff
@@ -2,40 +2,49 @@
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.crowncourt.dto.CrownCourtDTO;
+import uk.gov.justice.laa.crime.crowncourt.dto.maatcourtdata.RepOrderCCOutcomeDTO;
 import uk.gov.justice.laa.crime.crowncourt.model.ApiCrownCourtOutcome;
 import uk.gov.justice.laa.crime.crowncourt.model.ApiCrownCourtSummary;
-import uk.gov.justice.laa.crime.enums.CrownCourtOutcome;
+import uk.gov.justice.laa.crime.crowncourt.service.MaatCourtDataService;
+import uk.gov.justice.laa.crime.enums.CaseType;
 import uk.gov.justice.laa.crime.exception.ValidationException;
 
+import java.util.List;
 import java.util.Optional;
 
-import static java.util.Collections.emptyList;
-import static java.util.Optional.ofNullable;
-import static uk.gov.justice.laa.crime.enums.CrownCourtOutcome.CONVICTED;
-import static uk.gov.justice.laa.crime.enums.CrownCourtOutcome.PART_CONVICTED;
-
 @Slf4j
-@Component
+@Service
 @RequiredArgsConstructor
 public class CrownCourtDetailsValidator {
 
-    public static final String MSG_INVALID_CC_OUTCOME = "Imprisoned value cannot be null for Crown Court Outcome of Convicted or Partially Convicted";
-
-    public Optional<Void> validate(final ApiCrownCourtSummary crownCourtSummary) {
-        log.info("Validating crown court outcome with repId : {}", crownCourtSummary.getRepId());
+    private static final String CANNOT_HAVE_CROWN_COURT_OUTCOME_WITHOUT_MAGS_COURT_OUTCOME = "Cannot have Crown Court outcome without Mags Court outcome";
+    private static final String CHECK_CROWN_COURT_DETAILS_IMPRISONED_VALUE_MUST_BE_ENTERED_FOR_CROWN_COURT_OUTCOME = "Check Crown Court Details-Imprisoned value must be entered for Crown Court Outcome of ";
+    private static final String CONVICTED_PART_CONVICTED_REGEX = "CONVICTED|PART CONVICTED";
 
-        Optional<ApiCrownCourtOutcome> apiCrownCourtOutcome = ofNullable(crownCourtSummary.getCrownCourtOutcome())
-                .orElse(emptyList()).stream()
-                .filter(outcome -> outcome.getDateSet() == null)
-                .findFirst();
+    private final MaatCourtDataService maatCourtDataService;
 
-        if (apiCrownCourtOutcome.isPresent()) {
-            final CrownCourtOutcome crownCourtOutcome = apiCrownCourtOutcome.get().getOutcome();
-            log.info("Validating crown court outcome " + crownCourtOutcome);
-            if ((CONVICTED.getCode().equals(crownCourtOutcome.getCode()) || PART_CONVICTED.getCode().equals(crownCourtOutcome.getCode()))
-                    && crownCourtSummary.getIsImprisoned() == null) {
-                throw new ValidationException(MSG_INVALID_CC_OUTCOME);
+    public Optional<Void> checkCCDetails(CrownCourtDTO dto) {
+        ApiCrownCourtSummary crownCourtSummary = dto.getCrownCourtSummary();
+        List<RepOrderCCOutcomeDTO> repOrderCCOutcomeDTOList = maatCourtDataService.getRepOrderCCOutcomeByRepId(dto.getRepId(), null);
+        String caseType = dto.getCaseType() != null ? dto.getCaseType().getCaseType() : "X";
```

Comment:
I think this logic was recreated from the stored procedure, we dont need this. Instead, we can check if case type is not APPEAL_CC as below:
!CaseType.APPEAL_CC.getCaseType().equals(dto.getCaseType())

Reply:
Agree, caseType is a required field.